### PR TITLE
Remove upgrade scripts in favor of autoupgrade process

### LINKED ISSUE #23240: Move upgrade scripts to the autoupgrade module
#### Is your feature request related to a problem?

- The Core shouldn't be aware of the upgrade process.
- Putting the incremental upgrade scripts (SQL, PHP) in the core means that fixing them requires a new release.

#### Describe the solution you'd like

Everything related to upgrade should be removed from the core and handled by the module itself.
 
#### Additional context

This might require the module to stop behaving like a module, at least during the upgrade process.

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Remove the upgrade directory, everything has to be done by the autoupgrade module.
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes #23240
| How to test?      | CI is green.
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

## BC breaks
 The class `PrestaShopBundle\Install\Upgrade` no longer exists

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25794)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Review by matthieu-rolland — COMMENTED** [review]
phpstan issue, lgtm otherwise

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
I'll move all files to the autoupgrade modules (sync them), and add the migrateSettingsFile if needed

**Review by sowbiba — DISMISSED** [review]
Too many deleted files. I can't accept this :trollface: 

You have a conflict ;)

**Comment by PierreRambaud** [comment]
> Too many deleted files. I can't accept this :trollface:
> 
> You have a conflict ;)

Rebased 3 days ago, I'll wait for QA notification before rebasing again, and again

**Review by matks — DISMISSED** [review]

**Comment by Progi1984** [comment]
@PierreRambaud Could you rebase before sending it in QA ? Thanks

**Comment by PierreRambaud** [comment]
> @PierreRambaud Could you rebase before sending it in QA ? Thanks

As I said above, I'll wait for QA to be available, before rebasing again (already did a lot of time :sweat_smile: )

File: app/config/set_parameters.php [hunk 1]
```diff
@@ -27,14 +27,6 @@
 use PrestaShopBundle\Install\Upgrade;
 
 $parametersFilepath = __DIR__  . '/parameters.php';
-if (!file_exists($parametersFilepath)) {
-    // let's check first if there's some old config files which could be migrated
-    if (Upgrade::migrateSettingsFile() === false) {
```
Comment (jolelievre) [inline app/config/set_parameters.php:32]: Just a question, how is the `parameters.php` file generated now? From what I remember it was only created by this method, even if the name doesn't suggest it I believe it was the only code responsible for generating this file

**Review by jolelievre — DISMISSED** [review]
Thanks @PierreRambaud 

Maybe just indicate in the BC break description, that PrestaShop can no longer self-upgrade, especially with the CLI script which is documented here https://devdocs.prestashop.com/1.7/basics/keeping-up-to-date/upgrade/#how-to-upgrade-prestashop

We'll have to indicate this script is no longer available in the doc and invite people to rely on the autoupgrade module (until maybe a more generic and reusable library/repository is proposed)

File: app/config/set_parameters.php (hunk 1 above)
Comment (PierreRambaud) [reply app/config/set_parameters.php:32]: The file is generate in `config/bootstrap.php` this one is for old PrestaShop version and should be handle by the auotupgrade module. Not with a dirty check here :p

**Review by atomiix — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**